### ISSUE TITLE
v0.1.1 stalls and forfeits

### ISSUE THREAD

**Issue by tpoppins** [konsolas/ToppleChess#8]
Two of each observed in the ongoing 40/40 gauntlet

```
[Event "Topple 0.1.1 64-bit 40/40"]
[Site "Dual E5-2690v2"]
[Date "2018.06.28"]
[Round "7"]
[White "Topple 0.1.1 64-bit"]
[Black "Pirarucu 1.9.2 64-bit"]
[Result "1-0"]
[ECO "C00"]
[GameDuration "01:03:58"]
[GameEndTime "2018-06-28T03:23:38.779 Eastern Daylight Time"]
[GameStartTime "2018-06-28T02:19:40.007 Eastern Daylight Time"]
[Opening "French"]
[PlyCount "95"]
[Termination "stalled connection"]
[TimeControl "40/1200"]
[Variation "Labourdonnais Variation"]

1. e4 {book} e6 {book} 2. f4 {book} d5 {book} 3. exd5 {book} exd5 {book}
4. Nf3 {book} Nf6 {book} 5. d4 {-0.31/14 32s} Bb4+ {0.00/20 28s}
6. c3 {-0.19/15 32s} Bd6 {+0.03/22 32s} 7. Bd3 {-0.27/16 32s} c5 {+0.11/20 28s}
8. Qe2+ {+0.05/14 32s} Kf8 {+0.09/25 59s} 9. dxc5 {+0.36/16 32s}
Bxc5 {+0.10/27 26s} 10. Be3 {+0.32/15 32s} Bxe3 {+0.25/21 37s}
11. Qxe3 {+0.01/16 32s} Ng4 {+0.45/23 49s} 12. Qe2 {+0.41/16 32s}
Qc7 {+0.25/23 21s} 13. h3 {+0.38/15 32s} Nf6 {+0.09/23 30s}
14. Nbd2 {+0.23/15 32s} Qxf4 {+1.12/19 19s} 15. O-O {+0.21/15 32s}
Nc6 {+1.12/23 30s} 16. Rae1 {+0.05/15 30s} Qd6 {+1.14/24 29s}
17. b4 {+0.33/15 33s} Kg8 {+1.17/21 21s} 18. b5 {+0.71/15 33s}
Na5 {+1.02/21 27s} 19. Ng5 {+0.62/15 33s} Qc5+ {+0.98/24 19s}
20. Rf2 {+0.63/14 33s} Bd7 {+0.80/27 27s} 21. Qe7 {+0.48/15 33s}
Qxe7 {+0.87/27 15s} 22. Rxe7 {+0.69/17 33s} Be8 {+0.68/27 27s}
23. Nb3 {+0.73/15 33s} b6 {+0.72/26 23s} 24. Nxa5 {+1.56/16 31s}
bxa5 {-0.48/26 29s} 25. c4 {+1.61/16 33s} h6 {-0.56/25 24s}
26. Nf3 {+1.70/17 33s} Kf8 {-0.59/25 23s} 27. Rc7 {+1.66/17 32s}
h5 {-0.64/25 16s} 28. c5 {+2.14/16 30s} Rh6 {-0.47/23 21s}
29. Nd4 {+2.14/17 33s} Ne4 {-0.76/24 28s} 30. Bxe4 {+2.28/17 33s}
dxe4 {-1.44/26 16s} 31. Re2 {+2.15/17 33s} g5 {-1.48/24 11s}
32. Rxe4 {+2.41/16 33s} a4 {-1.85/24 14s} 33. Nf5 {+2.29/16 33s}
Re6 {-2.91/26 12s} 34. Rxe6 {+3.04/18 33s} fxe6 {-3.16/31 14s}
35. Nd6 {+3.25/18 31s} e5 {-3.55/26 16s} 36. b6 {+6.39/22 31s}
axb6 {-8.44/33 15s} 37. cxb6 {+6.11/21 34s} Ra6 {-9.03/35 12s}
38. b7 {+6.45/20 34s} Rb6 {-9.42/34 9.3s} 39. Nxe8 {+6.89/19 33s}
Rb1+ {-9.87/36 16s} 40. Kf2 {+7.50/18 34s} Kxe8 {-10.78/36 16s}
41. Rc8+ {+8.04/19 29s} Kd7 {-12.62/65 54s} 42. b8=Q {+7.99/17 30s}
Rxb8 {-13.46/68 38s} 43. Rxb8 {+8.54/18 29s} Kd6 {-14.36/64 35s}
44. Rg8 {+9.27/18 30s} g4 {-14.93/76 50s} 45. hxg4 {+9.65/17 30s}
hxg4 {-19.55/75 43s} 46. Rxg4 {+9.60/18 30s} a3 {-38.31/96 51s}
47. Ra4 {+11.48/20 33s} Kd7 {-40.85/93 39s}
48. Ke3 {+10.48/18 30s, Black's connection stalls} 1-0

```
```
[Event "Topple 0.1.1 64-bit 40/40"]
[Site "Dual E5-2690v2"]
[Date "2018.06.27"]
[Round "6"]
[White "CT800 1.30 64-bit"]
[Black "Topple 0.1.1 64-bit"]
[Result "1-0"]
[ECO "B23"]
[GameDuration "00:27:31"]
[GameEndTime "2018-06-27T22:44:06.160 Eastern Daylight Time"]
[GameStartTime "2018-06-27T22:16:34.491 Eastern Daylight Time"]
[Opening "Sicilian"]
[PlyCount "43"]
[Termination "stalled connection"]
[TimeControl "40/1200"]
[Variation "Closed"]

1. e4 {book} c5 {book} 2. Nc3 {book} d6 {book} 3. Bc4 {book} Nc6 {book}
4. Nge2 {book} e5 {book} 5. a3 {+0.13/11 33s} Be7 {-0.18/16 32s}
6. O-O {+0.01/13 25s} Nf6 {-0.16/16 32s} 7. d3 {-0.01/12 34s} O-O {-0.18/15 32s}
8. Be3 {-0.01/13 23s} Ng4 {+0.02/16 32s} 9. Bd2 {-0.01/13 21s}
Bg5 {-0.06/17 32s} 10. Nd5 {+0.01/13 38s} Bxd2 {-0.24/15 29s}
11. Qxd2 {+0.01/6 0.088s} Be6 {-0.16/16 33s} 12. b4 {+0.01/13 39s}
Re8 {-0.30/14 33s} 13. bxc5 {+0.03/12 43s} dxc5 {-0.19/16 33s}
14. Nec3 {+0.01/12 30s} Nd4 {-0.23/16 33s} 15. Rfb1 {+0.04/12 25s}
Qh4 {0.00/14 33s} 16. h3 {+0.72/13 26s} Nf3+ {0.00/16 33s}
17. gxf3 {+3.21/13 13s} Qxh3 {0.00/18 33s} 18. fxg4 {+5.04/6 0.091s}
Qxg4+ {0.00/16 33s} 19. Kh2 {-0.30/18 22s} Qh5+ {0.00/18 33s}
20. Kg1 {-0.30/17 25s} Qg4+ {0.00/19 33s} 21. Kh2 {-0.30/20 15s}
Qh5+ {0.00/19 33s} 22. Kg1 {-0.30/19 19s, Black's connection stalls} 1-0

```
```
[Event "Topple 0.1.1 64-bit 40/40"]
[Site "Dual E5-2690v2"]
[Date "2018.06.28"]
[Round "8"]
[White "Topple 0.1.1 64-bit"]
[Black "BigLion 2.23x"]
[Result "0-1"]
[ECO "A40"]
[GameDuration "00:36:26"]
[GameEndTime "2018-06-28T04:33:04.063 Eastern Daylight Time"]
[GameStartTime "2018-06-28T03:56:37.098 Eastern Daylight Time"]
[Opening "Queen's pawn"]
[PlyCount "78"]
[Termination "time forfeit"]
[TimeControl "40/1200"]

1. d4 {book} e6 {book} 2. Nf3 {book} f5 {book} 3. g3 {book} Nf6 {book}
4. Bg2 {book} d5 {book} 5. Nc3 {+0.15/15 32s} Be7 {-0.25/11 31s}
6. Bf4 {+0.07/14 32s} Nh5 {-0.21/11 31s} 7. Bd2 {+0.12/16 32s}
Nc6 {-0.16/11 31s} 8. Ne5 {+0.06/16 32s} Nxe5 {-0.24/11 31s}
9. dxe5 {+0.56/15 32s} d4 {-0.24/10 14s} 10. Nb5 {+0.70/16 30s}
c6 {-0.25/11 31s} 11. e3 {+0.93/16 31s} dxe3 {+0.02/11 30s}
12. fxe3 {+0.91/16 33s} Nxg3 {+0.02/10 8.1s} 13. hxg3 {+0.92/16 33s}
cxb5 {+0.02/9 6.9s} 14. Qh5+ {+1.25/17 31s} g6 {-0.06/10 16s}
15. Qxg6+ {+1.32/17 33s} hxg6 {-0.14/11 7.4s} 16. Rxh8+ {+1.17/18 33s}
Bf8 {-0.14/11 10s} 17. Bb4 {+1.51/18 33s} Kf7 {-0.17/11 36s}
18. Rxf8+ {+1.55/18 33s} Qxf8 {-0.23/13 5.7s} 19. Bxf8 {+1.50/20 33s}
Kxf8 {-0.67/12 17s} 20. O-O-O {+1.61/20 33s} Ke7 {-0.41/12 39s}
21. Rh1 {+1.49/20 33s} Rb8 {-0.96/12 39s} 22. Rh6 {+1.51/19 33s}
Kf7 {-0.91/13 38s} 23. Rh7+ {+1.77/20 33s} Kg8 {-0.27/14 38s}
24. Rc7 {+1.90/20 33s} Kf8 {-0.32/14 38s} 25. e4 {+1.86/19 33s}
a6 {-0.37/14 38s} 26. a3 {+2.11/20 33s} Kg8 {-0.38/15 38s}
27. Bf3 {+2.32/20 33s} Kf8 {-0.40/14 11s} 28. Kb1 {+2.17/19 33s}
Kg8 {-0.43/15 40s} 29. exf5 {+2.41/22 33s} gxf5 {-0.97/15 40s}
30. g4 {+2.42/22 33s} fxg4 {-0.53/16 40s} 31. Bxg4 {+2.51/22 33s}
Kf8 {-0.59/14 20s} 32. Bh5 {+2.51/21 33s} b6 {-0.58/14 41s}
33. Bf3 {+2.80/22 33s} a5 {-1.36/14 41s} 34. Ra7 {+3.20/22 31s}
b4 {-1.80/16 41s} 35. axb4 {+3.68/24 33s} axb4 {-2.30/15 41s}
36. Ka2 {+4.30/24 39s} Ke8 {-3.40/16 41s} 37. Bc6+ {+4.60/24 42s}
Kd8 {-3.52/16 11s} 38. Kb3 {+6.21/26 29s} b5 {-3.55/15 12s}
39. Kxb4 {+8.12/26 29s} Rb6 {-3.99/14 25s, White loses on time} 0-1

```
```
[Event "Topple 0.1.1 64-bit 40/40"]
[Site "Dual E5-2690v2"]
[Date "2018.06.28"]
[Round "11"]
[White "Topple 0.1.1 64-bit"]
[Black "Aice 0.99.2"]
[Result "0-1"]
[ECO "A45"]
[GameDuration "01:15:50"]
[GameEndTime "2018-06-28T13:27:45.898 Eastern Daylight Time"]
[GameStartTime "2018-06-28T12:11:55.845 Eastern Daylight Time"]
[Opening "Queen's pawn game"]
[PlyCount "156"]
[Termination "time forfeit"]
[TimeControl "40/1200"]

1. d4 {book} Nf6 {book} 2. g3 {book} d5 {book} 3. Bg2 {book} Bf5 {book}
4. c4 {book} c6 {book} 5. Nf3 {+0.21/15 30s} e6 {-0.44/12 32s}
6. Nc3 {+0.26/15 32s} dxc4 {-0.20/11 20s} 7. Ne5 {+0.30/16 29s}
Nbd7 {-0.23/12 39s} 8. Nxc4 {+0.39/16 30s} Nb6 {-0.28/12 32s}
9. Nxb6 {+0.21/15 33s} Qxb6 {+0.17/11 4.5s} 10. e4 {+0.13/15 33s}
Bg6 {-0.04/12 24s} 11. e5 {+0.19/15 33s} Ng8 {+0.02/12 38s}
12. Qb3 {+0.59/16 33s} Qxb3 {+0.71/14 38s} 13. axb3 {+0.33/16 32s}
Bc2 {+0.75/13 52s} 14. Ne4 {+0.32/15 33s} Bxb3 {+0.73/12 22s}
15. Bf4 {+0.52/15 33s} a6 {+0.83/11 23s} 16. O-O {+0.20/15 33s}
O-O-O {+1.15/13 60s} 17. Be3 {+0.01/16 33s} Bb4 {+1.19/12 20s}
18. Rfc1 {+0.05/16 33s} Ne7 {+1.31/12 13s} 19. Ng5 {0.00/17 33s}
Rdf8 {+1.30/13 21s} 20. Ne4 {-0.08/15 30s} h6 {+1.27/12 16s}
21. Nc5 {+0.19/15 33s} Bd5 {+1.01/14 45s} 22. Nd3 {-0.02/16 33s}
Bxg2 {+1.59/13 23s} 23. Kxg2 {-0.03/16 31s} Nd5 {+1.54/13 19s}
24. Rc2 {-0.09/16 33s} Rd8 {+1.59/12 21s} 25. b3 {-0.18/16 33s}
Be7 {+1.76/12 37s} 26. Bd2 {-0.13/16 33s} Nc7 {+1.76/13 28s}
27. Be3 {-0.14/15 31s} Nb5 {+1.86/13 25s} 28. Rc4 {-0.26/15 33s}
Rd5 {+1.86/11 21s} 29. Nf4 {-0.26/15 33s} Rd7 {+1.84/13 26s}
30. Ra2 {-0.33/15 33s} h5 {+1.88/11 18s} 31. Ne2 {-0.57/16 33s}
Bd8 {+1.89/12 28s} 32. Raa4 {-0.75/16 33s} Bb6 {+1.88/11 20s}
33. Nc3 {-0.74/15 33s} Nxc3 {+1.99/11 13s} 34. Rxc3 {-0.87/18 33s}
Rd5 {+1.92/12 21s} 35. Kh1 {-1.10/16 33s} h4 {+2.07/12 13s}
36. Kg1 {-1.33/18 33s} hxg3 {+2.06/12 24s} 37. hxg3 {-1.72/18 33s}
Kc7 {+2.16/11 10s} 38. Kg2 {-1.53/17 33s} Rh7 {+2.06/12 21s}
39. Rc1 {-1.28/17 33s} Rh5 {+2.11/11 14s} 40. g4 {-1.23/18 31s}
Rh7 {+2.10/11 3.2s} 41. Kg3 {-1.22/18 30s} Rb5 {+2.04/13 56s}
42. Rb1 {-1.22/17 29s} Ba7 {+1.95/14 100s} 43. f4 {-1.28/17 30s}
g6 {+2.01/14 81s} 44. Kg2 {-1.14/17 30s} Bb6 {+2.06/13 49s}
45. Kg3 {-1.13/19 30s} Rh8 {+2.07/13 44s} 46. Rc4 {-1.18/19 30s}
Rd5 {+2.06/12 42s} 47. Ra4 {-1.16/18 28s} Rhd8 {+2.17/12 26s}
48. Rd1 {-1.20/19 30s} Kb8 {+2.30/13 21s} 49. Kf3 {-1.14/18 30s}
c5 {+2.43/14 44s} 50. Bf2 {-1.14/17 30s} cxd4 {+2.43/13 25s}
51. Rd3 {-1.14/17 30s} Bc5 {+2.40/13 60s} 52. Ke4 {-1.03/18 30s}
Ba7 {+2.33/12 22s} 53. Rc4 {-1.04/18 30s} Bb6 {+2.32/12 22s}
54. g5 {-0.94/18 30s} Rc8 {+2.50/13 38s} 55. Rb4 {-0.90/18 30s}
Bc5 {+2.49/14 37s} 56. Rc4 {-1.09/19 27s} Ba7 {+2.53/13 13s}
57. Rxc8+ {-1.05/20 30s} Kxc8 {+2.46/5 0.006s} 58. Rd1 {-1.38/22 30s}
Bb6 {+2.76/14 46s} 59. Kf3 {-1.42/20 30s} Kd7 {+2.75/15 46s}
60. Rd3 {-1.55/21 30s} Bc5 {+2.78/15 44s} 61. Ke4 {-1.37/20 30s}
a5 {+2.80/14 22s} 62. Be1 {-1.39/21 30s} b6 {+2.76/15 40s}
63. Bd2 {-1.77/20 27s} Ke7 {+2.79/15 38s} 64. Be1 {-1.40/20 29s}
Kd8 {+2.74/15 49s} 65. Bd2 {-1.62/20 31s} Kc7 {+2.80/15 17s}
66. Be1 {-1.62/21 31s} Rd8 {+2.74/14 29s} 67. Bf2 {-1.55/20 28s}
Kc6 {+2.81/14 15s} 68. Be1 {-1.71/21 31s} Kb5 {+2.82/14 22s}
69. Bf2 {-1.68/22 31s} Ra8 {+2.93/14 17s} 70. Rf3 {-1.50/21 31s}
a4 {+3.47/16 16s} 71. bxa4+ {-2.18/22 32s} Rxa4 {+3.74/15 18s}
72. Rd3 {-2.20/20 31s} Ra2 {+3.65/16 18s} 73. Bxd4 {-3.05/21 31s}
Ra4 {+5.02/14 18s} 74. Kf3 {-4.35/22 31s} Rxd4 {+5.71/14 12s}
75. Rc3 {-5.29/21 31s} Kc6 {+5.71/13 16s} 76. Kg4 {-5.90/21 31s}
Kd5 {+6.36/15 28s} 77. Rc1 {-6.14/21 31s} Bb4 {+6.55/14 46s}
78. Rc7 {-5.36/22 31s} Bd2 {+7.70/14 19s, White loses on time} 0-1

```

Topple-ivybridge.exe
Dual E5-2690v2 (IvyBridge, 20 cores) @3 GHz, 32 GB RAM, HT off
Win 7 x64 Pro
Cute Chess GUI 2018-06-25 x64 build
concurrency=16

No illegal moves, though, that appears to have been fixed.
On the other hand, I raised the average strength of the lineup for this gauntlet (as compared to the aborted one for v0.1.0) by 50 Elo and Topple still outstrips the field by 100 Elo. That's some improvement, stalls and forfeits notwithstanding!
Tirsa@CCRL

**Comment by konsolas** [comment]
This partially seems to be a problem with repetition detection, where if there have nearly been 3 repetitions, Topple won't stop searching. That explains the round 6 forfeit, but not any of the others.

**Comment by konsolas** [comment]
In the first game you posted @tpoppins (round 7), it appears that it was Pirarucu 1.9.2 which lost on time, and not Topple.

**Comment by konsolas** [comment]
Ok, better synopsis: 

- 2nd game posted was due to the repetition bug
- 3rd and 4th games posted were lost on time just before the time controls repeated, so I might need a larger buffer there.

**Comment by konsolas** [comment]
Should be fixed by 0c0e25a923567064e9b4cc8f02ff8ab28eb9a1a4 (release v0.1.2)

Thank you for testing :)
